lib/src/native/realm_core.dart

```diff
@@ -1123,10 +1165,10 @@ extension on Pointer<IntPtr> {
 }
 
 extension on Pointer<Void> {
-  T? toObject<T extends Object>() {
+  T? toObject<T extends Object>([bool isPersistent = false]) {
     assert(this != nullptr, "Pointer<Void> is null");
 
-    final object = _realmLib.gc_handle_to_object(this);
+    Object object = isPersistent ? _realmLib.persistent_handle_to_object(this) : _realmLib.weak_handle_to_object(this);
```

Comment:
I prefer the previous approach, where you don't have to distinguish.

Reply:
I refactored it since persistence handles are different than weak handles. Do you see an issue with the current approach?

Reply:
Also not sure what you mean by "don't have to distinguish". Previously we had the same distinction in another place

Reply:
Previously `gc_handle_deref` worked for both weak and hard handles